From 526bf199ed8900945f9dffc3041ec322a758285c Mon Sep 17 00:00:00 2001
From: Hans Rosenfeld <hans.rosenfeld@joyent.com>
Date: Wed, 11 Sep 2019 13:50:10 +0000
Subject: [PATCH] OS-7981 mpt_sas hangs after config header request timeout
 Reviewed by: Robert Mustacchi <rm+illumos@fingolfin.org> Reviewed by: Jerry
 Jelinek <jerry.jelinek@joyent.com> Approved by: Jerry Jelinek
 <jerry.jelinek@joyent.com>

---
 usr/src/uts/common/io/scsi/adapters/mpt_sas/mptsas_impl.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/usr/src/uts/common/io/scsi/adapters/mpt_sas/mptsas_impl.c b/usr/src/uts/common/io/scsi/adapters/mpt_sas/mptsas_impl.c
index 87abb767ad..c4a856b2bf 100644
--- a/usr/src/uts/common/io/scsi/adapters/mpt_sas/mptsas_impl.c
+++ b/usr/src/uts/common/io/scsi/adapters/mpt_sas/mptsas_impl.c
@@ -24,7 +24,7 @@
  * Copyright 2012 Nexenta Systems, Inc. All rights reserved.
  * Copyright 2014 OmniTI Computer Consulting, Inc. All rights reserved.
  * Copyright (c) 2014, Tegile Systems Inc. All rights reserved.
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2019, Joyent, Inc.
  */
 
 /*
@@ -378,6 +378,7 @@ mptsas_access_config_page(mptsas_t *mpt, uint8_t action, uint8_t page_type,
 	 * Check if the header request completed without timing out
 	 */
 	if (cmd->cmd_flags & CFLAG_TIMEOUT) {
+		config_flags |= MPTSAS_CMD_TIMEOUT;
 		mptsas_log(mpt, CE_WARN, "config header request timeout");
 		rval = DDI_FAILURE;
 		goto page_done;
@@ -520,6 +521,7 @@ mptsas_access_config_page(mptsas_t *mpt, uint8_t action, uint8_t page_type,
 	 * Check if the request completed without timing out
 	 */
 	if (cmd->cmd_flags & CFLAG_TIMEOUT) {
+		config_flags |= MPTSAS_CMD_TIMEOUT;
 		mptsas_log(mpt, CE_WARN, "config page request timeout");
 		rval = DDI_FAILURE;
 		goto page_done;
-- 
2.21.0

